stdlib/sql/sql_test.go

```diff
@@ -0,0 +1,355 @@
+						{int64(6)},
+						{int64(1)},
+						{int64(643)},
+						{int64(42)},
+						{int64(1283)},
+						{int64(4)},
+						{int64(0)},
+						{int64(18)},
+					},
+				},
+			},
+		},
+		{
+			name:   "tinyints",
+			column: "_tinyint",
+			value:  intSliceToInterface([]int64{6, 1, 643, 42, 1283, 4, 0, 18}),
+			want: []*executetest.Table{
+				{
+					ColMeta: []flux.ColMeta{
+						{Label: "_tinyint", Type: flux.TInt},
+					},
+					Data: [][]interface{}{
+						{int64(6)},
+						{int64(1)},
+						{int64(643)},
+						{int64(42)},
+						{int64(1283)},
+						{int64(4)},
+						{int64(0)},
+						{int64(18)},
+					},
+				},
+			},
+		},
+		{
+			name:   "strings",
+			column: "_string",
+			value:  stringSliceToInterface([]string{"6", "1", "643", "42", "1283", "4", "0", "18"}),
+			want: []*executetest.Table{
+				{
+					ColMeta: []flux.ColMeta{
+						{Label: "_string", Type: flux.TString},
+					},
+					Data: [][]interface{}{
+						{"6"},
+						{"1"},
+						{"643"},
+						{"42"},
+						{"1283"},
+						{"4"},
+						{"0"},
+						{"18"},
+					},
+				},
+			},
+		},
+		{
+			name:   "times",
+			column: "_id",
+			value:  createTestTimes(),
+			want: []*executetest.Table{
+				{
+					ColMeta: []flux.ColMeta{
+						{Label: "_id", Type: flux.TTime},
+					},
+					Data: [][]interface{}{
+						{values.ConvertTime(createTestTimes()[0].(time.Time))},
+						{values.ConvertTime(createTestTimes()[1].(time.Time))},
+						{values.ConvertTime(createTestTimes()[2].(time.Time))},
+						{values.ConvertTime(createTestTimes()[3].(time.Time))},
+						{values.ConvertTime(createTestTimes()[4].(time.Time))},
+						{values.ConvertTime(createTestTimes()[5].(time.Time))},
+						{values.ConvertTime(createTestTimes()[6].(time.Time))},
+						{values.ConvertTime(createTestTimes()[7].(time.Time))},
+					},
+				},
+			},
+		},
+	}
+
+	db, mock, err := sqlmock.NewWithDSN(dsn)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			// Setup mock database
+			// We expect two queries because of the way that we compare TableObjects to Tables (see functions starting at line 211)
+			expectedQuery := fmt.Sprintf("SELECT %s FROM TestTable", tc.column)
+
+			rs := sqlmock.NewRows([]string{tc.column})
+			for _, v := range tc.value {
+				rs.AddRow(v)
+			}
+
+			mock.ExpectQuery(expectedQuery).WillReturnRows(rs)
+			mock.ExpectQuery(expectedQuery).WillReturnRows(rs)
+
+			// 'from' function
+			fromSQLSignature := semantic.FunctionPolySignature{
+				Parameters: map[string]semantic.PolyType{
+					"driverName":     semantic.String,
+					"dataSourceName": semantic.String,
+					"query":          semantic.String,
+				},
+				Required: semantic.LabelSet{"driverName", "dataSourceName", "query"},
+				Return:   flux.TableObjectType,
+			}
+
+			fluxFn := flux.FunctionValue(FromSQLKind, createFromSQLOpSpec, fromSQLSignature).Function()
```

Comment:
instead of calling the function from this level,  let's instantiate a  SQLIterator type directly, call `connect` `fetch` and `decode` ,  then compare the table returned by `decode` to the expected result.